src/Response/Handler/ResponseHandler.php

```diff
@@ -18,21 +19,22 @@ class ResponseHandler implements ResponseHandlerInterface
     /**
      * @param ResponseInterface $response
      *
-     * @return array
+     * @return ResponseData
      *
      * @throws NotFoundResponseException
      * @throws BadRequestResponseException
      * @throws UnexpectedResponseException
      * @throws JsonException
      */
-    public function handle(ResponseInterface $response): array
+    public function handle(ResponseInterface $response): ResponseData
     {
         $statusCode = $response->getStatusCode();
         $body       = $response->getBody();
+        $headers    = $response->getHeaders();
 
         if ($statusCode >= 200 && $statusCode < 300) {
             if ($body === null || (int)$body->getSize() === 0) {
-                return [];
+                return new ResponseData([], $headers);
```

Comment:
Could we add the status code property to the class? I see a number of ways how it could be useful later on.

- [ ] add status code property

Reply:
It's available